pyriemann/embedding.py

```diff
@@ -92,13 +111,293 @@ def fit_transform(self, X, y=None):
+        return X_new
+
+    def fit_transform(self, X, y=None):
+        """Calculate the coordinates of the embedded points.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y : ndarray | None (default: None)
+            Not used, here for compatibility with sklearn API.
+
+        Returns
+        -------
+        X_: ndarray, shape (n_matrices, n_components)
+            Coordinates of embedded matrices.
+        """
+        self.fit(X)
+        return self.embedding_
+
+
+def barycenter_weights(X, Y, indices, metric='riemann', reg=1e-3):
+    """Compute Riemannian barycenter weights of X from Y along the first axis.
+
+    Estimates the weights to assign to each point in Y[indices] to recover
+    the point X[i] by geodesic interpolation. The barycenter weights sum to 1.
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_matrices, n_channels, n_channels)
+        Set of SPD matrices.
+    Y : ndarray, shape (n_matrices, n_channels, n_channels)
+        Set of SPD matrices.
+    indices : ndarray, shape (n_matrices, n_neighbors)
+        Indices of the points in Y used to compute the barycenter
+    metric : {'riemann', 'logeuclid', 'euclid'}, default: 'riemann'
+        Kernel metric.
+    reg : float, default=1e-3
+        amount of regularization to add for the problem to be
+        well-posed in the case of n_neighbors > n_dim
+
+    Returns
+    -------
+    B : ndarray, shape (n_matrices, n_neighbors)
+        Interpolation weights.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+    See sklearn.manifold._locally_linear.barycenter_weights for original code.
+    """
+    n_matrices, n_neighbors = indices.shape
+    assert X.shape[0] == n_matrices, "Number of index-sets in indices must " \
+                                     "match number of matrices in X."
+
+    B = np.empty((n_matrices, n_neighbors), dtype=X.dtype)
+    v = np.ones(n_neighbors, dtype=X.dtype)
+
+    for i in range(n_matrices):
+        X_neighbors = Y[indices[i]]
+        G = kernel(X_neighbors, Cref=X[i], metric=metric)
+        trace = np.trace(G)
+        if trace > 0:
+            R = reg * trace
+        else:
+            R = reg
+        G.flat[:: n_neighbors + 1] += R
+        w = solve(G, v, sym_pos=True)
+        B[i] = w / np.sum(w)
+    return B
+
+
+def locally_linear_embedding(X,
+                             *,
+                             n_components=2,
+                             n_neighbors=5,
+                             metric='riemann',
+                             reg=1e-3):
+    """Riemannian Locally Linear Embedding (LLE).
+
+    Riemannian LLE as proposed in [1]_. LLE is a non-linear, neighborhood-
+    preserving dimensionality reduction algorithm which consists of three
+    main steps:
+    For each point x,
+    1.  find its k nearest neighbors KNN(x) and
```

Comment:
Like scikit-learn, docstring of class `LocallyLinearEmbedding` must be different from function `locally_linear_embedding`.
```suggestion
    """Perform a Locally Linear Embedding (LLE) of SPD matrices. 

    As proposed in [1]_, Locally Linear Embedding (LLE) is a non-linear, neighborhood-
    preserving dimensionality reduction algorithm which consists of three
    main steps. For each point x,

    1.  find its k nearest neighbors KNN(x) and
```